daemons/gptp/common/ptp_message.cpp

```diff
@@ -1467,6 +1469,8 @@ void PTPMessagePathDelayRespFollowUp::processMessage(IEEE1588Port * port)
 		 remote_req_rx_timestamp.nanoseconds);
 
 	// Adjust turn-around time for peer to local clock rate difference
+	// TODO: Are these .998 and 1.002 specifically defined in the standard?
```

Comment:
I couldn't find anything relating to a threshold for whether to do a rate ratio adjustment of turnaround time. According to 11.2.15.2.4 of 802.1AS, we should always scale `link_delay` by the neighbor rate ratio rather than `turn_around`, at least if I'm understanding the code and spec correctly.

Regardless of whether we're scaling the right term, I think that scaling only when the rate ratio is outside a range is likely to introduce an exaggerated step-change when the range boundary is crossed. It would probably be good to find the author of the code and ask what it's meant to do, though, if that's possible.